apps/src/netsim/NetSimRouterLogModal.js

```diff
@@ -148,28 +171,190 @@ NetSimRouterLogModal.prototype.render = function () {
   }));
   this.rootDiv_.html(renderedMarkup);
 
+  // Add input handlers
   this.getRouterLogToggleButton().one('click', function() {
     this.toggleRouterLogMode_();
     this.render();
   }.bind(this));
 
-  this.rootDiv_.on('shown.bs.modal.onModalOpen', function () {
-    if (this.shard_) {
-      this.shard_.logTable.subscribe();
-    }
-  }.bind(this));
-
-  this.rootDiv_.on('hidden.bs.modal.onModalClose', function () {
-      if (this.shard_) {
-        this.shard_.logTable.unsubscribe();
-      }
-  }.bind(this));
-
   this.rootDiv_.find('th').click(function (event) {
     this.onSortHeaderClick_($(event.target).attr('data-sort-key'));
   }.bind(this));
+
+  // Add rows to the table
+  var rows = this.getSortedFilteredLogEntries(this.logEntries_)
+      .slice(0, MAXIMUM_ROWS_IN_FULL_RENDER)
+      .map(this.makeTableRow_.bind(this));
+  this.rootDiv_.find('tbody').append(rows);
+
+  if (rows.length === MAXIMUM_ROWS_IN_FULL_RENDER) {
+    var maxRenderedWarning = document.createElement('div');
+    maxRenderedWarning.className = 'log-browser-limit-message';
+    maxRenderedWarning.innerText = i18n.showingFirstXLogEntries({
+      x: MAXIMUM_ROWS_IN_FULL_RENDER
+    });
+    this.rootDiv_.find('table').after(maxRenderedWarning);
+  }
+};
+
+/**
+ * Convert the given set of log entries to table rows and insert them
+ * into the DOM, instead of re-rendering the whole table.
+ * @param {!NetSimLogEntry[]} newEntries
+ * @private
+ */
+NetSimRouterLogModal.prototype.renderNewLogEntries_ = function (newEntries) {
+  // Be lazy, don't render at all if not visible.
+  if (!this.isVisible() || newEntries.length === 0) {
+    return;
+  }
+
+  /** @type {jQuery} Table body element. */
+  var tbody = this.rootDiv_.find('tbody');
+
+  // Get existing table row elements, which are already sorted and filtered.
+  var oldRows = tbody.find('tr');
+
+  // Sort and filter the new entries, and generate DOM rows for them.
+  newEntries = this.getSortedFilteredLogEntries(newEntries);
+  var newRows = $(newEntries.map(this.makeTableRow_.bind(this)));
+
+  // Get the current sort value function
+  var iterateeFunction = NetSimRouterLogModal.iterateeMap[this.sortBy_];
```

Comment:
iteratee is an unconventional name, I first thought it was a  typo.  Can you come up with something more descriptive of what is being iterated?